### ISSUE TITLE
Upgrade Header Validation Case Sensitivity

### ISSUE THREAD

**Issue by max-saccoccio-bg** [miguelgrinberg/python-engineio#203]
The change introduced to fix #72 added validation that incoming connections have a correct "UPGRADE" header supplied.

It appears as if this validation is case sensitive, which appears to be at odds with IETF RFC 6455: https://tools.ietf.org/html/rfc6455#section-4.2.1 item 5:

```
   3.   An |Upgrade| header field containing the value "websocket",
        treated as an ASCII case-insensitive value.
```

We are creating a client in a test environment which is supplying "WebSocket" for the value of this header and we are getting failed connections and the "invalid transport" error on our server:

```
Invalid transport websocket (further occurrences of this error will be logged with level INFO)
```

Are we doing something wrong with our client or does the validation that was introduced need to be case insensitive?

**Comment by komputerwiz** [comment]
Perfect timing! Ran into the same problem earlier this morning. I just submitted a pull request (#204) addressing this issue.

**Comment by miguelgrinberg** [comment]
Fixed by #204
